scanpy/get/_aggregated.py

```diff
@@ -0,0 +1,486 @@
+        Data matrix for aggregation.
+    weight
+        Weights to be used for aggergation.
+    key_set
+        Subset of keys to which to filter.
+    """
+
+    groupby: pd.Series
+    data: Array
+    weight: pd.Series | Array
+    key_set: Set[str] | None
+
+    def __post_init__(self):
+        if self.key_set is not None and not isinstance(self.key_set, Set):
+            self.key_set = dict.fromkeys(self.key_set).keys()
+
+    def count(self, *, _indices: Indices | None = None) -> np.ndarray:
+        """\
+        Count the number of observations in each group.
+
+        Returns
+        -------
+        Array of counts.
+        """
+        if _indices is None:
+            _indices = self._extract_indices()
+        return np.bincount(_indices.key_index)
+
+    def sum(self) -> Array:
+        """\
+        Compute the sum per feature per group of observations.
+
+        Returns
+        -------
+        Array of sum.
+        """
+        A, _ = self._sparse_aggregator(normalize=False)
+        return utils.asarray(A * self.data)
+
+    def mean(self, *, _A: spmatrix | None = None) -> Array:
+        """\
+        Compute the mean per feature per group of observations.
+
+        Returns
+        -------
+        Array of mean.
+        """
+        if _A is None:
+            _A, _ = self._sparse_aggregator(normalize=True)
+        return utils.asarray(_A @ self.data)
+
+    def count_mean_var(self, dof: int = 1, *, _indices: Indices | None = None) -> CMV:
+        """\
+        Compute the count, as well as mean and variance per feature, per group of observations.
+
+        The formula `Var(X) = E(X^2) - E(X)^2` suffers loss of precision when the variance is a
+        very small fraction of the squared mean. In particular, when X is constant, the formula may
+        nonetheless be non-zero. By default, our implementation resets the variance to exactly zero
+        when the computed variance, relative to the squared mean, nears limit of precision of the
+        floating-point significand.
+
+        Params
+        ------
+        dof
+            Degrees of freedom for variance.
+
+        Returns
+        -------
+        Object with `count`, `mean`, and `var` attributes.
+        """
+        assert dof >= 0
+        if _indices is None:
+            _indices = self._extract_indices()
+        A, _ = self._sparse_aggregator(normalize=True, _indices=_indices)
+        count_ = self.count(_indices=_indices)
+        mean_ = self.mean(_A=A)
+        # sparse matrices do not support ** for elementwise power.
+        mean_sq = utils.asarray(A @ _power(self.data, 2))
+        if self.weight is None:
+            sq_mean = mean_**2
+        else:
+            A_unweighted, _ = Aggregate(
+                groupby=self.groupby,
+                data=self.data,
+                weight=self.weight,  # TODO: why pass weights when creating unweighted A?
+                key_set=self.key_set,
+            )._sparse_aggregator()
```

Comment:
So the difference to above is that it doesn’t call `_sparse_aggregator` with `normalize=True`. So this should probably be simply:

```suggestion
            A_unweighted, _ = self._sparse_aggregator()
```